sky/skylet/providers/gcp/node_provider.py

```diff
@@ -229,6 +231,21 @@ def terminate_node(self, node_id: str):
                         ),
                     )
                     result = resource.stop_instance(node_id=node_id)
+
+                    # Check if the instance is actually stopped.
+                    # GCP does not fully stop an instance even after the stop operation is finished.
+                    for _ in range(MAX_POLLS_STOP):
+                        instance = resource.get_instance(node_id=node_id)
+                        if instance.is_stopped():
+                            logger.info(f'Instance {node_id} is stopped.')
+                            break
+                        elif instance.is_stopping():
+                            time.sleep(POLL_INTERVAL)
+                        else:
+                            raise RuntimeError(f'Unexpected instance status. Details: {instance}')
+
+                    if instance.is_stopping():
```

Comment:
...is still in STOPPING status.

Reply:
Sorry I don't quite get this comment?

Reply:
Update error message on 248: "is still not terminated" -> "is still in STOPPING status"

Reply:
Got it.